### PR TITLE
Week 4 andrew luna

### PR DESCRIPTION
**Pull request by lundrew**

### CODE REVIEW

File: code-challenges.test.js [hunk 1]
```diff
@@ -15,35 +15,125 @@
...
+// // return the new randomized array
 
+const remFirstShuff = (array) => {
 
-// b) Create the function that makes the test pass.
+    array.shift()  
+
+    for (let i = array.length - 1; i > 0; i--) {
+        const j = Math.floor(Math.random() * (i + 1));
+        [array[i], array[j]] = [array[j], array[i]];
```
Comment (WalkerTexas-coder) [inline code-challenges.test.js:50]: Good use of destructuring!

File: code-challenges.test.js [hunk 2]
```diff
@@ -15,35 +15,125 @@
...
+// // Expected output example (can be a different order): ["saffron", "aquamarine", "periwinkle", "indigo", "ochre"]
 
 
-// b) Create the function that makes the test pass.
+describe('remFirstShuff', () => {
+    const colors1 = ["purple", "blue", "green", "yellow", "pink"]
+    const colors2 = ["chartreuse", "indigo", "periwinkle", "ochre", "aquamarine", "saffron"]
+    it('takes in an array, removes the first item from the array and shuffles the remaining content', () => {
+      expect(remFirstShuff(colors1)).toEqual(expect.arrayContaining(colors1));
+      expect(remFirstShuff(colors2)).toEqual(expect.arrayContaining(colors2));
```
Comment (WalkerTexas-coder) [inline code-challenges.test.js:29]: Well done!

File: code-challenges.test.js [hunk 3]
```diff
@@ -15,35 +15,125 @@
...
+    })
 
-// a) Create a test with an expect statement using the variables provided.
+    var min = array.reduce(function(a, b) {
+        return Math.min(a, b) 
+    })
+
+    return [min, max] 
+
+}
```
Comment (WalkerTexas-coder) [inline code-challenges.test.js]: Looking up what Math.min / .max do would really help with the refactoring process.

File: code-challenges.test.js [hunk 4]
```diff
@@ -15,35 +15,125 @@
...
+    let uniqueChars = [];
+
+combined.forEach((element) => {
+    if (!uniqueChars.includes(element)) {
+        uniqueChars.push(element);
+    }
+
+});
+    return uniqueChars
+}
```
Comment (WalkerTexas-coder) [inline code-challenges.test.js:148]: Very cool! Just missing the pseudo code for this one!

File: ruby-challenges.rb [hunk 5]
```diff
@@ -22,6 +35,24 @@
...
+    i = 0
+    while i < str.length
+      if vowels.include?(str[i])
+        string_array[i] =  nil
+      end
+      i +=1
+    end
+
+    return string_array.join
+end
```
Comment (WalkerTexas-coder) [inline ruby-challenges.rb:50]: Very cool!

File: interview-questions.md [hunk 6]
```diff
@@ -7,52 +7,60 @@ Then, research the question to expand on your answer. Even if you feel you have
...
-  Researched answer:
+  Researched answer: DOM events are signals that something has occurred or can be triggered by user interactions or by the browser. 
+  Examples of DOM events include:
+  - When a user clicks the mouse
+  - When a web page has loaded
+  - When an image has been loaded
+  - When the mouse moves over an element
+  - When an input field is changed
+  - When an HTML form is submitted
+  - When a user presses a key
```
Comment (WalkerTexas-coder) [inline interview-questions.md:28]: Good work!

**Review by WalkerTexas-coder — COMMENTED** [review]
Well done very cool use of different tools! You are missing pseudo code on code challenge #3. Post up that pseudo code and we'll get you approved.

**Review by WalkerTexas-coder — CHANGES_REQUESTED** [review]
Well done very cool use of different tools! You are missing pseudo code on code challenge #3. Post up that pseudo code and we'll get you approved.

File: code-challenges.test.js (hunk 4 above)
Comment (lundrew) [reply code-challenges.test.js:148]: All done!

**Review by WalkerTexas-coder — APPROVED** [review]
Awesome work man! Thank you for the pseudo code!

**Pull request merged**